lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1733,6 +1733,65 @@ Error ONNXModelLoader::loadArgMax(const ONNX_NAMESPACE::NodeProto &op,
   return Error::success();
 }
 
+Error ONNXModelLoader::loadUpsample(const ONNX_NAMESPACE::NodeProto &op,
+                                    ArgumentDictionaryTy &dict) {
+
+  RETURN_ERR_IF_NOT(opsetVersion_ < 10,
+                    "Upsample Operator is deprecated for Opset 10 and above");
+
+  const std::string &opName = loadOperatorName(op);
+  NodeValue in;
+  ASSIGN_VALUE_OR_RETURN_ERR(in, getNodeValueByName(op.input(0)));
+  std::string mode;
+  if (dict.count("mode")) {
+    ASSIGN_VALUE_OR_RETURN_ERR(mode, loadStr(dict.at("mode")));
+  }
+
+  /// Only Nearest Mode is supported
+  RETURN_ERR_IF_NOT(mode.compare("nearest") == 0,
+                    "Upsample Operator has nearest mode support only");
+
+  /// Scale is always float as per onnx documentation
+  std::vector<float> scales;
+
+  RETURN_ERR_IF_NOT(opsetVersion_ > 6,
+                    "Upsample Operator is supported for Opset 7 and above");
+
+  if (opsetVersion_ == 7) {
+    if (dict.count("scales")) {
```

Comment:
What if `opsetVersion_ == 7 && !dict.count("scales")`? We'd never initialize `scales`. What default values should be used in this case?

Reply:
https://github.com/onnx/onnx/blob/master/docs/Changelog.md#Upsample-7 does not mention default values and it has the field as required. So whoever generates the onnx file when they run check_model API then the API should fail

Reply:
Ok, but to be safe we should also check and fail here. Instead of `if (dict.count("scales"))` can you simply do `RETURN_ERR_IF_NOT(dict.count("scales"), "scales is required for Upsample");` or something?